### ISSUE TITLE
add peek support

### ISSUE THREAD

**Issue by web3creator** [datafuselabs/openraft#935]
**Is your feature request related to a problem? Please describe.**
add 
```
pub async fn peek_async(&self)->result<(),error>;
pub  fn peek(&self)->result<(),error>;

**Describe the solution you'd like**
Node and master node remain synchronized

**Describe alternatives you've considered**
The underlying framework determines whether the current node and the master node are synchronized with the master node at this moment.

**Comment by AdamLeyshon** [comment]
Isn't that what [ensure_linearizable](https://docs.rs/openraft/0.9.11/openraft/raft/struct.Raft.html#method.ensure_linearizable) is for?

If I understand it right, this function ensures that the node is up to date with Raft logs?

**Comment by drmingdrmer** [comment]
> Isn't that what [ensure_linearizable](https://docs.rs/openraft/0.9.11/openraft/raft/struct.Raft.html#method.ensure_linearizable) is for?
> 
> If I understand it right, this function ensures that the node is up to date with Raft logs?

Yes It looks like `peek()` is just forward `ensure_linearizable` to leader on the follower.
